docs/reference/setup/cluster_restart.asciidoc

```diff
@@ -30,55 +33,108 @@ PUT _cluster/settings
 +
 --
 
-Shard recovery will be much faster if you stop indexing and issue a
-<<indices-synced-flush, synced-flush>> request:
+Performing a <<indices-synced-flush, synced-flush>> speeds up shard
+recovery.
 
 [source,sh]
 --------------------------------------------------
 POST _flush/synced
 --------------------------------------------------
 // CONSOLE
 
-A synced flush request is a ``best effort'' operation. It will fail if there
-are any pending indexing operations, but it is safe to reissue the request
-multiple times if necessary.
+Synced flush operations are performed on a "best effort" basis. The request
+fails if there are any pending indexing operations, but it is safe to
+reissue the request multiple times until it succeeds.
 --
 
-. *Shutdown and upgrade all nodes*
+. *Shutdown all nodes.*
 +
 --
+* If you are running Elasticsearch with `systemd`:
++
+[source,sh]
+--------------------------------------------------
+sudo systemctl stop elasticsearch.service
+--------------------------------------------------
 
-Stop all Elasticsearch services on all nodes in the cluster. Each node can be
-upgraded following the same procedure described in <<upgrade-node>>.
+* If you are running Elasticsearch with SysV `init`:
++
+[source,sh]
+--------------------------------------------------
+sudo -i service elasticsearch stop
+--------------------------------------------------
+
+* If you are running Elasticsearch as a daemon:
++
+[source,sh]
+--------------------------------------------------
+kill `cat pid`
+--------------------------------------------------
 --
 
-. *Upgrade any plugins*
+. *Upgrade all nodes.*
 +
 --
+To upgrade using a <<deb,Debian>> or <<rpm,RPM>> package:
+
+*   Use `rpm` or `dpkg` to install the new package.  All files are
+    installed in the appropriate location for the operating system
+    and Elasticsearch config files are not overwritten.
+
+To upgrade using a zip or compressed tarball:
 
-Elasticsearch plugins must be upgraded when upgrading a node.  Use the
-`elasticsearch-plugin` script to install the correct version of any plugins
-that you need.
+.. Extract the zip or tarball to a _new_ directory to be sure that you don't
+    overwrite the `config` or `data` directories.
+
+.. Either copy the files in the `config` directory from your old installation
+    to your new installation, or set the environment variable `ES_JVM_OPTIONS`
+    to the location of the `jvm.options` file and use the `-E path.conf=`
+    option on the command line to point to your external config directory.
+
+.. Either copy the files in the `data` directory from your old installation
+    to your new installation, or set `path.data` in `config/elasticsearch.yml`
+    to point to your external data directory.
+
+[TIP]
+================================================
+
+When using the zip or tarball packages, the `config`, `data`, `logs` and
+`plugins` directories are placed within the Elasticsearch home directory by
```

Comment:
"home directory" is pretty overloaded. I think it'd be more clear to day that those directories come in the zip but we recommend moving them. Frankly, I recommend using the rpm or deb for real deployments, but I'm sure there are folks that can't.